lib/uplinkc/testdata/object_test.c

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+#include <string.h>
+#include <stdlib.h>
+
+#include "require.h"
+#include "uplink.h"
+#include "helpers.h"
+
+void handle_project(ProjectRef project);
+
+int main(int argc, char *argv[]) {
+    ProjectOptions opts = {};
+    memset(&opts.key, '\0', 32);
+    memcpy(&opts.key, "hello", 5);
+
+    with_test_project(&handle_project, &opts);
+}
+
+void handle_project(ProjectRef project) {
+    char *_err = "";
+    char **err = &_err;
+
+    char *bucket_name = "TestBucket";
+
+    BucketConfig config = test_bucket_config();
+    BucketInfo info = create_bucket(project, bucket_name, &config, err);
+    require_noerror(*err);
+    free_bucket_info(&info);
+
+    EncryptionAccess access = {};
+    memset(&access.key, '\0', 32);
+    memcpy(&access.key, "hello", 5);
+
+    BucketRef bucket = open_bucket(project, bucket_name, access, err);
+    require_noerror(*err);
+
+    char *object_paths[] = {"TestObject1","TestObject2","TestObject3","TestObject4"};
+    int num_of_objects = 4;
+
+    // NB: about +500 years from time of writing
+    int64_t future_expiration_timestamp = 17329017831;
+
+    for(int i = 0; i < num_of_objects; i++) {
+        int data_len = 1024 * pow((i + 1), 2);
+        char *data = mkrndstr(data_len);
+
+        { // upload
+            MapRef map = new_map_ref();
+            UploadOptions opts = {
+                "text/plain",
+                map,
+                future_expiration_timestamp,
+            };
+
+            UploaderRef uploader = upload(bucket, object_paths[i], &opts, err);
```

Comment:
Wondering whether it should match the libuplink. We have currently few namings for this thing, mobile uplink is using NewWriter (to be consistent with GCS).